e2e/main_test.go

```diff
@@ -16,27 +17,41 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-// Integration tests are run against memstore whereas.
-// Testnetintegration tests are run against eigenda backend talking to testnet disperser.
+// Integration tests are run against memstore whereas
+// testnet integration tests are run against eigenda backend talking to testnet disperser.
 // Some of the assertions in the tests are different based on the backend as well.
 // e.g, in TestProxyServerCaching we only assert to read metrics with EigenDA
 // when referencing memstore since we don't profile the eigenDAClient interactions
 var (
 	runTestnetIntegrationTests bool // holesky tests
-	runIntegrationTests        bool // memstore tests
+	runIntegrationTests        bool // memstore V1 tests
+	runIntegrationTestsV2      bool // memstore V2 tests
 	runFuzzTests               bool // fuzz tests
 )
 
+func flagActivated(envVar string) bool {
+	return os.Getenv(envVar) == "true" || os.Getenv(envVar) == "1"
+}
+
 // ParseEnv ... reads testing cfg fields. Go test flags don't work for this library due to the dependency on Optimism's E2E framework
 // which initializes test flags per init function which is called before an init in this package.
 func ParseEnv() {
-	runFuzzTests = os.Getenv("FUZZ") == "true" || os.Getenv("FUZZ") == "1"
-	runIntegrationTests = os.Getenv("INTEGRATION") == "true" || os.Getenv("INTEGRATION") == "1"
-	runTestnetIntegrationTests = os.Getenv("TESTNET") == "true" || os.Getenv("TESTNET") == "1"
+	runFuzzTests = flagActivated("FUZZ")
+	runIntegrationTestsV2 = flagActivated("INTEGRATION_V2")
+	runIntegrationTests = flagActivated("INTEGRATION")
+	runTestnetIntegrationTests = flagActivated("TESTNET")
 
 	if runIntegrationTests && runTestnetIntegrationTests {
 		panic("only one of INTEGRATION=true or TESTNET=true env var can be set")
 	}
+
+	if runIntegrationTests && runIntegrationTestsV2 {
+		panic("only one of INTEGRATION=true or INTEGRATION_V2=true env var can be set")
```

Comment:
why is this the case? why can't I run both v1 and v2? Is it because there are race conditions between the two? should be documented if so

Reply:
I don't know why it is like this, @ethenotethan can you chime in here?

Reply:
Did this ensure testmodes were uniquely set. Otherwise atm someone could provide true for both and it'd just use V2 so making them mutually exclusive seemed most practical. We'll probably need to revisit this once we start testing migrations though

Reply:
Yeah, this will change once we implement the changes necessary for canary testing: there will no longer be an "either-or" choice between v1 and v2